ratis-grpc/src/main/java/org/apache/ratis/grpc/server/GrpcServerProtocolService.java

```diff
@@ -316,7 +316,7 @@ boolean isHeartbeat(AppendEntriesRequestProto request) {
 
       @Override
       String requestToString(AppendEntriesRequestProto request) {
-        return ServerStringUtils.toAppendEntriesRequestString(request);
+        return ServerStringUtils.toAppendEntriesRequestString(request, e -> "");
```

Comment:
Let's pass `null`.  The code currently can handle `null`.

BTW, could you also update `stateMachineLogEntryProtoToString(..)` to print only sizes and 
`toLogEntryString(..)` to use `stateMachineLogEntryProtoToString(..)`?
```java
@@ -70,7 +70,11 @@ public final class LogProtoUtils {
   }
 
   static String stateMachineLogEntryProtoToString(StateMachineLogEntryProto p) {
-    return "logData:" + p.getLogData() + ", stateMachineEntry:" + p.getType() + ":" + p.getStateMachineEntry();
+    final StateMachineEntryProto stateMachineEntry = p.getStateMachineEntry();
+    return p.getType()
+        + ": logData.size=" + p.getLogData().size()
+        + ", stateMachineData.size=" + stateMachineEntry.getStateMachineData().size()
+        + ", logEntryProtoSerializedSize=" + stateMachineEntry.getLogEntryProtoSerializedSize();
   } 
```

```java
@@ -46,9 +45,10 @@ public final class LogProtoUtils {
     }
     final String s;
     if (entry.hasStateMachineLogEntry()) {
-      s = ", " + Optional.ofNullable(function)
-          .orElseGet(() -> proto -> "" + ClientInvocationId.valueOf(proto))
-          .apply(entry.getStateMachineLogEntry());
+      if (function == null) {
+        function = LogProtoUtils::stateMachineLogEntryProtoToString;
+      }
+      s = ", " + function.apply(entry.getStateMachineLogEntry());
     } else if (entry.hasMetadataEntry()) {
       final MetadataProto metadata = entry.getMetadataEntry();
       s = "(c:" + metadata.getCommitIndex() + ")";
```